src/main/java/io/jenkins/pluginhealth/scoring/probes/SpotBugsProbe.java

```diff
@@ -0,0 +1,66 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+
+
+@Component
+@Order(SpotBugsProbe.ORDER)
+public class SpotBugsProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SpotBugsProbe.class);
+    public static final int ORDER = ContinuousDeliveryProbe.ORDER + 1;
+    public static final String KEY = "spotbugs";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        if (plugin.getDetails().get(SCMLinkValidationProbe.KEY) == null) {
+            LOGGER.error("Couldn't run {} on {} because previous SCMLinkValidationProbe has null value in database", key(), plugin.getName());
+            return ProbeResult.error(key(), "SCM link has not been validated yet");
+        }
+    final Path repository = context.getScmRepository();
+    try (Stream<Path> paths = Files.find(repository, 2,
+                (file, basicFileAttributes) -> Files.isReadable(file)
+                        && ("pom.xml".equalsIgnoreCase(file.getFileName().toString())))) {
+            Optional<Path> pomFile = paths.findFirst();
```

Comment:
This would discriminate plugins using `Gradle` as build tool.

Reply:
I was under impression that the Jenkins plugin uses only maven as a build tool. Is it only the majority that uses maven?